pkg/agent/proxy/proxier.go

```diff
@@ -558,9 +534,15 @@ func (p *proxier) installServices() {
 				}
 			}
 
+			var hasNestedService bool
+			// Check the `NestedServiceSupport` field only when Proxy is enabled with `supportNestedService`.
+			if p.supportNestedService {
+				hasNestedService = svcInfo.NestedServiceSupport
+			}
+
 			// Install ClusterIP flows for the Service.
 			groupID := p.groupCounter.AllocateIfNotExist(svcPortName, internalPolicyLocal)
-			if err := p.ofClient.InstallServiceFlows(groupID, svcInfo.ClusterIP(), uint16(svcInfo.Port()), svcInfo.OFProtocol, uint16(affinityTimeout), externalPolicyLocal, corev1.ServiceTypeClusterIP); err != nil {
+			if err := p.ofClient.InstallServiceFlows(groupID, svcInfo.ClusterIP(), uint16(svcInfo.Port()), svcInfo.OFProtocol, uint16(affinityTimeout), externalPolicyLocal, corev1.ServiceTypeClusterIP, hasNestedService); err != nil {
```

Comment:
I think the new parameter should be a pointer

- nil, do nothing in `InstallServiceFlows`. The Service is not related to nested Service, like NodePort or LoadBalancer.
- true, install extra flow in EndpointDNATTable. The Service could be used as Endpoint of MC Service.
- false, add extra action NestedServiceRegMark to the flow in ServiceLBTable. The Service is a MC Service.

In addition, this pointer should be always nil when MC is not enabled.

Reply:
I remember that you said at least now  only ClusterIP can be used as the Endpoint of MC Service according to Kubernetes KEP-xxxx. If so, I think we should only enhance the flow of pure ClusterIP Service and skip the ClusterIP in NodePort or LoadBalancer.

Reply:
Yes, I mean `ClusterIP`, not ClusterIP type of Service, since both NodePort and LoadBalancer have ClusterIP, I mean ClusterIP can be exported instead of `NodeIP:NodePort` or "LBIP:Port". KEP-1645 also suggested to use ClusterIP of NodePort and LoadBalancer Services.

Reply:
Could we support that using ClusterIP of NodePort or LoadBalancer as Endpoint of MC-Service currently?